src/WireMock.Net/Server/WireMockServer.Admin.cs

```diff
@@ -555,11 +555,63 @@ private ResponseMessage MappingsPost(RequestMessage requestMessage)
 
         private ResponseMessage MappingsDelete(RequestMessage requestMessage)
         {
-            ResetMappings();
+            // if we allow body for HTTP Delete & a body is defined
+            if (_settings.AllowBodyForAllHttpMethods.HasValue && _settings.AllowBodyForAllHttpMethods.Value && !String.IsNullOrEmpty(requestMessage.Body))
```

Comment:
For admin commands, it's not needed to check for `AllowBodyForAllHttpMethods`. It's ok to use this functionality.